### DISCUSSION TITLE
Question about client-side translations

### DISCUSSION THREAD

**Discussion by MellKam** [Alexandre-Fernandez/astro-i18n#62] [General]
Hello and thank you for your work!

I tried your package and noticed something. In the output script tag for client-side translations, there are translations for all locales, even if we only need one. Is there a reason for this? It might not be very efficient, especially with lots of locales.

Can we make it better by including only the translations needed for the current page? This could make things faster and save space on the client's side.

**Comment by Alexandre-Fernandez** [comment]
The reason this was implemented like this is because I needed to cover most use-cases and some use-cases include having client-side logic. Such as switching locales on the client-side or fetching translations that were not ran in the server.

Your concern is still valid though, and I thought about implementing that while developing v2.0. I have however limited free time and I'll be mainly do maintenance work on the library for now.

The workaround would be to use server-mode and handle the hydration yourself. Otherwise try to have most translations come from your page translations.

In the future maybe a config option could be added to specify which translations to serialize.

By the way there's the same problem with route segments.

**Comment by MellKam** [comment]
@Alexandre-Fernandez Why did you close the issue and discussion if the problem is not solved? I think it would be useful for other people to see that this problem exists.

**Comment by Alexandre-Fernandez** [comment]
You asked why this was done like this and I replied, if there's nothing more to talk there's no need for this to be opened. Also this is not a problem but a feature request and I listed it in the [to-do list](https://github.com/Alexandre-Fernandez/astro-i18n/discussions/63). As you mentioned the discussion is closed not deleted, people can still see it.

**Reply by MellKam** [reply]
Sorry, I didn't see the todo list you created
